File: test/test-compound-nostring.lua
-- hdf5-udf <file.h5> test-compound-nostring.lua Temperature.lua:1000:double

function dynamic_dataset()
    local compound = lib.getData("Dataset1")
    local udf_data = lib.getData("Temperature.lua")
    local udf_dims = lib.getDims("Temperature.lua")
    local ffi = require("ffi")

    for i=0, udf_dims[1]-1 do
        print(string.format(
            "serial: %d, temperature: %.6f, pressure: %.6f",
            tonumber(compound[i].serial_number),
            tonumber(compound[i].temperature),
            tonumber(compound[i].pressure)))
        udf_data[i] = compound[i].temperature
    end
end

File: test/test-compound-nostring.py
# hdf5-udf <file.h5> test-compound-nostring.py Temperature.py:1000:double
def dynamic_dataset():
    compound = lib.getData("Dataset1")
    udf_data = lib.getData("Temperature.py")
    udf_dims = lib.getDims("Temperature.py")

    for i in range(udf_dims[0]):
        print("serial: {}, temperature: {:.6f}, pressure: {:.6f}".format(
            compound[i].serial_number,
            compound[i].temperature,
            compound[i].pressure), flush=True)
        udf_data[i] = compound[i].temperature
